File: python/937.py
from typing import List

class Solution(object):
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        digit_list, letter_list = [], []

        for log in logs:
            if log.split()[1].isdigit():
                digit_list.append(log)
            if log.split()[1].isalpha():
                letter_list.append(log)

        letter_list.sort(key=lambda x: (x.split()[1:], x.split()[0]))

        return letter_list + digit_list


if __name__ == "__main__":
    solution = Solution()

    log_list: List[str] = ["dig1 8 1 5 1", "let1 art can", "dig2 3 6", "let2 own kit dig", "let3 art zero"]

    print(solution.reorderLogFiles(log_list))

File: cpp/937.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<string> splitBySpace(string input) {
    string delimiter = " ";
    vector<string> splitVec;

    size_t pos = 0;
    while ((pos = input.find(delimiter)) != string::npos) {
        splitVec.push_back(input.substr(0, pos));
        input.erase(0, pos + delimiter.length());
    }

    return splitVec;
}

vector<string> reorderLogFiles(vector<string> &logs) {
    vector<string> letters;
    vector<string> digits;

    for (auto log : logs) {
        if (isdigit(splitBySpace(log)[1][0])) digits.push_back(log);
        else letters.push_back(log);
    }

    sort(letters.begin(),
         letters.end(),
         [](string leftLog, string rightLog) {
             string delimiter = " ";
             size_t pos = leftLog.find(delimiter);
             string leftStrIdentifier = leftLog.substr(0, pos);
             string leftStrWOIdentifier = leftLog.substr(pos + 1, string::npos);
             pos = rightLog.find(delimiter);
             string rightStrIdentifier = rightLog.substr(0, pos);
             string rightStrWOIdentifier = rightLog.substr(pos + 1, string::npos);

             return (leftStrWOIdentifier == rightStrWOIdentifier ? leftStrIdentifier < rightStrIdentifier : leftStrWOIdentifier < rightStrWOIdentifier);
         });

    vector<string> output;
    for (auto str : letters) {
        output.push_back(str);
    }

    for (auto str : digits) {
        output.push_back(str);
    }

    return output;
}

int main() {

    // vector<string> logs = {"dig1 8 1 5 1","let1 art zero","dig2 3 6","let2 own kit dig","let3 art zero"};
    vector<string> logs = {"2 A","1 B","4 C","1 A"};

    vector<string> output = reorderLogFiles(logs);
    for (auto str : output) {
        cout << str << endl;
    }

    return 0;
}